### ISSUE TITLE
CAL-226: Support installing to existing LVM volumes

### ISSUE THREAD

**Issue by calamares-user** [calamares/calamares#427]
Jira issue originally created by user edwardoo:

There is quite a bit of demand for LVM support, in part because:

* LVM excels for more complex, multi-boot setups, and Calamares is used mainly by "smaller" distros, often installed beside "bigger" ones.
* LVM is setup by Fedora's default partitioning scheme.
* Once LVM is setup, it is a bit hard to shrink it to make room for non-lvm distros

Possible simplifications for a first version which would solve most of the complaints:

* Support install to existing volumes only (people can use existing distro to do the setup)
* Supporting it for / (root) only  (home can be added to the fstab later)
* Only support already mounted logical volumes

Basically, the idea is to have a possibility for experienced people to install Calamares-based distros when they have lvm. User-friendly support can come later.

PS: I just thought this should be tracked/prioritized here somewhere. Sorry if I'm wrong.

**Comment by jcfrosty** [comment]
3.1.1 released, but still no true LVM support?

**Comment by stikonas** [comment]
@jcfrosty there are now various pull requests that should make partitioning operations work on LVM LV volumes
https://github.com/calamares/calamares/pull/846
https://github.com/calamares/calamares/pull/828
After they land, other Calamares modules (probably bootloader and initramfs) will still need some work to add LVM support.

And lastly, LVM VG operations will have to be added, but that's beyond what this bug wants.

**Comment by stikonas** [comment]
I've now successfully installed onto preexisting LVM volume group (logical volumes were created by the installer).

**Comment by philmmanjaro** [comment]
Will test also on my end soon. The current CAL in **unstable** has the needed patches and is based on **kpmcore v3.3**.

**Comment by RafalSkolasinski** [comment]
Is it known when this functionality will roll out to distros using Calamares installer? 
At least installing on created by hand LVM volumes.

**Comment by stikonas** [comment]
@RafalSkolasinski It should already be available. kpmcore 3.3. has now propagated to most distributions. Calamares more recently gained support for creating/removing LVM VGs too but maybe it's not yet in most distros.

**Comment by RafalSkolasinski** [comment]
I tried Solus installation not more than a week ago and installer didn't see any of my preexisting LVM volumes.
It was not a problem of linux system itself or driver as ``lvs`` was showing all of them properly. 

Anyway, I really hope to see this feature out there soon :).

**Comment by stikonas** [comment]
Maybe lvm was not installed on the live system?

**Comment by RafalSkolasinski** [comment]
No, I did see my lvm groups and volumes from the live system terminal.




--
Rafał Skolasiński
Postdoctoral Researcher, QuTech
Delft University of Technology Delft, The Netherlands

GitHub: https://github.com/RafalSkolasinski
Kwant GitLab: https://gitlab.kwant-project.org/r-j-skolasinski

**Comment by RafalSkolasinski** [comment]
Oh, you're probably right. I must had got confused and think that it is calamares it is using. 
I am sorry to give you troubles! 

I believe Manjaro and Sabayon are using Calamares? 
I will be testing those two soon, with lvm setup. I will let you know if I will encounter any problems there.

**Comment by Pointedstick** [comment]
If this feature has been implemented, shouldn't the issue be closed?

**Comment by stikonas** [comment]
> If this feature has been implemented, shouldn't the issue be closed?

Yes, it should. When I added support to installing to existing LVM volumes, I still thought that we should keep it open until LVM VG operations are implemented. They were implemented in GSoC this year (althogh, I haven't tested Calamares after that). So now it should be closed. But I don't have permissions to close the bug.

**Comment by Pointedstick** [comment]
Me neither. @adriaandegroot?

**Comment by adriaandegroot** [comment]
LVM is something I leave entirely to people who know about it (e.g. @stikonas and @cjlcarvalho) and who want it. I have neither quality. If this is implemented (with latest kpmcore) and someone says so, I'll close the ticket.

**Comment by philmmanjaro** [comment]
Normally it can be closed. However, I see the following patches not yet included in **v3.2.2**:

- [partition] Fixing LVM LV name line edit visibility. - committed on 13 Aug
- [partition] Don't show erase button for inactive RAID. - committed on 13 Aug
- [partition] Disabling partition buttons for inactive RAID. - committed on 13 Aug

**Comment by stikonas** [comment]
Well, there are more potential bugs (e.g. #888 is it still there?) But we can probably track them separately from this bug. No software is bug free anyway.

**Comment by philmmanjaro** [comment]
OK. I'm fine with closing this one. @adriaandegroot: maybe you add @stikonas as an CAL developer so he may support you in closing issues as needed.

**Comment by adriaandegroot** [comment]
I'm marking it as Milestone 3.2.3 to indicate that it's available in v 3.2.3 .. those commits @philmmanjaro  mentions are in master, waiting for the next release (which is waiting on the swap-UI stuff being finished).

**Comment by philmmanjaro** [comment]
@stikonas: for #888 I didn't had time yet to test the issue further. Personally I'm not using LVM at all. @adriaandegroot: alright. I'm already merging master to my packages anyway. It is always good for others to know and keep track on ongoing development :smiley:

**Comment by cjlcarvalho** [comment]
I was planning to work on #888 some time ago. I can take a look at it.
